fmf/base.py

```diff
@@ -588,3 +590,60 @@ def copy(self):
         duplicate = copy.deepcopy(self)
         self.parent = duplicate.parent = original_parent
         return duplicate
+
+    def _get_raw_data_node(self):
+        """
+        Get tuple
+          - node where _raw_data are stored
+          - list if keys for current node inside raw data
+        """
+        raw_data_node = self
+        id_list = list()
+        while True:
+            raw_data = raw_data_node._raw_data
+            if raw_data:
+                return raw_data_node, id_list
+            elif not raw_data_node.parent:
+                raise utils.RootError("Cannot find root node with _raw_data, very strange")
+            else:
+                id_list.insert(0, "/" + raw_data_node.name.rsplit("/")[-1])
+                raw_data_node = raw_data_node.parent
+
+    def _get_node_raw_data(self):
+        """
+        Get dictionary of _raw_data for current node
+        """
+        raw_data_node, id_list = self._get_raw_data_node()
+        raw_data = raw_data_node._raw_data
+        for key in id_list:
+            raw_data = raw_data[key]
+        return raw_data
+
+    def modify(self, data, method="update"):
+        """
+        Modify node metadata (original data stored in file) via method.
+        method is any method what contains python dictionary (default: update)
+        and data is attribute of the method
+
+        Modification will invalide affected node data and childs, please reload the Tree
+
+        Be aware that it modify raw data in file:
+        If you have defined "key+: fist" in the file for node and
+        you will add "key: second" it will lead  to "secondfirst" output for node
+
+        Data are always stored into the last sourced fmf file of the modified node
+          (e.g no inheritance and no elasticity)
+        """
+
+        getattr(self._get_node_raw_data(), method)(data)
```

Comment:
I've run into the following traceback:

```python
>>> fmf.Tree('examples/wget').find('/download/requirements/spider').modify(dict(x=1))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/psss/git/fmf/fmf/base.py", line 636, in modify
    getattr(self._get_node_raw_data(), method)(data)
AttributeError: 'NoneType' object has no attribute 'update'
```

Are you able to reproduce as well?

Reply:
Good catch @psss it is caused by no data inside the dict element. I've expected to be empty dict, but it is None.   So that I have to init dict there.